### PR TITLE
UI: Add progress bar for re-packaging shaders

### PR DESCRIPTION
**Pull request by nreinartz**
Addresses request #4105

Uses the already existing progress bar but swaps out the label for the one proposed in the issue.

### CODE REVIEW

**Comment by MutantAura** [comment]
Seems to work well for me 👍

File: src/Ryujinx.Ava/Assets/Locales/en_US.json [hunk 1]
```diff
@@ -562,6 +562,7 @@
   "LoadingHeading": "Loading {0}",
   "CompilingPPTC": "Compiling PTC",
   "CompilingShaders": "Compiling Shaders",
+  "PackagingShaders": "Packaging Shaders",
```
[the remark this answers is not in the document]
Comment (nreinartz) [reply src/Ryujinx.Ava/Assets/Locales/en_US.json]: Makes sense, fixed that.

File: src/Ryujinx.Graphics.Gpu/Shader/DiskCache/ParallelDiskCacheLoader.cs [hunk 2]
```diff
@@ -299,19 +299,23 @@ public void LoadShaders()
...
-                        foreach (var kv in _programList)
+                        for (int i = 0; i < _programList.Count; i++)
                         {
                             if (!Active)
                             {
                                 break;
                             }
 
-                            (CachedShaderProgram program, byte[] binaryCode) = kv.Value;
+                            (CachedShaderProgram program, byte[] binaryCode) = _programList.Values[i];
```
Comment (AcK77) [inline src/Ryujinx.Graphics.Gpu/Shader/DiskCache/ParallelDiskCacheLoader.cs:315]:
```suggestion
                            (CachedShaderProgram program, byte[] binaryCode) = _programList.Values[i];

```

File: src/Ryujinx.Graphics.Gpu/Shader/DiskCache/ParallelDiskCacheLoader.cs [hunk 3]
```diff
@@ -299,19 +299,25 @@ public void LoadShaders()
...
                     {
+                        _stateChangeCallback(ShaderCacheState.Packaging, 0, _programList.Count);
+
                         Logger.Info?.Print(LogClass.Gpu, $"Rebuilding {_programList.Count} shaders...");
 
                         using var streams = _hostStorage.GetOutputStreams(_context);
 
-                        foreach (var kv in _programList)
+                        int packagedShaders = 0;
+                        foreach(var kv in _programList)
```
Comment (gdkchan) [inline src/Ryujinx.Graphics.Gpu/Shader/DiskCache/ParallelDiskCacheLoader.cs:309]:
```suggestion
                        foreach (var kv in _programList)
```

**Review by marysaka — APPROVED** [review]

**Review by MutantAura — APPROVED** [review]

**Pull request merged**